zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/TenantAppliersTest.java

```diff
@@ -0,0 +1,170 @@
+
+import io.camunda.zeebe.engine.state.mutable.MutableAuthorizationState;
+import io.camunda.zeebe.engine.state.mutable.MutableProcessingState;
+import io.camunda.zeebe.engine.state.mutable.MutableTenantState;
+import io.camunda.zeebe.engine.state.mutable.MutableUserState;
+import io.camunda.zeebe.engine.util.ProcessingStateExtension;
+import io.camunda.zeebe.protocol.impl.record.value.tenant.TenantRecord;
+import io.camunda.zeebe.protocol.impl.record.value.user.UserRecord;
+import io.camunda.zeebe.protocol.record.value.AuthorizationOwnerType;
+import io.camunda.zeebe.protocol.record.value.AuthorizationResourceType;
+import io.camunda.zeebe.protocol.record.value.EntityType;
+import io.camunda.zeebe.protocol.record.value.PermissionType;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(ProcessingStateExtension.class)
+public class TenantAppliersTest {
+
+  private MutableProcessingState processingState;
+
+  private MutableTenantState tenantState;
+  private MutableUserState userState;
+  private MutableAuthorizationState authorizationState;
+  private TenantDeletedApplier tenantDeletedApplier;
+  private TenantEntityAddedApplier tenantEntityAddedApplier;
+
+  @BeforeEach
+  public void setup() {
+    tenantState = processingState.getTenantState();
+    userState = processingState.getUserState();
+    authorizationState = processingState.getAuthorizationState();
+    tenantDeletedApplier =
+        new TenantDeletedApplier(
+            processingState.getTenantState(),
+            processingState.getUserState(),
+            processingState.getAuthorizationState());
+    tenantEntityAddedApplier =
+        new TenantEntityAddedApplier(
+            processingState.getTenantState(), processingState.getUserState());
+  }
+
+  @Test
+  void shouldAddEntityToTenant() {
+    // given
+    final long entityKey = UUID.randomUUID().hashCode();
+    final long tenantKey = UUID.randomUUID().hashCode();
+    final var tenantId = UUID.randomUUID().toString();
+    createTenant(tenantKey, tenantId);
+    createUser(entityKey, "username");
+    authorizationState.insertOwnerTypeByKey(tenantKey, AuthorizationOwnerType.TENANT);
+    // when
+    associateUserWithTenant(tenantKey, tenantId, entityKey);
+    // then
+    assertThat(tenantState.getEntitiesByType(tenantKey).get(EntityType.USER))
+        .containsExactly(entityKey);
+    final var persistedUser = userState.getUser(entityKey).get();
+    assertThat(persistedUser.getTenantIdsList()).containsExactly(tenantId);
+  }
+
+  @Test
+  void shouldDeleteTenantWithMultipleUsers() {
+    // given
+    final long tenantKey = UUID.randomUUID().hashCode();
+    final var tenantId = UUID.randomUUID().toString();
+
+    final var tenantRecord = createTenant(tenantKey, tenantId);
+    authorizationState.insertOwnerTypeByKey(tenantKey, AuthorizationOwnerType.TENANT);
```

Comment:
❌ Please remove this from the test and make sure we do this in the `TenantCreatedApplier`. Look into the [RoleCreatedApplier](https://github.com/camunda/camunda/blob/main/zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/RoleCreatedApplier.java#L31) for inspiration. We need to make sure this CF gets populated and it appears I missed this during the review of the `TenantCreatedApplier`.

If we fix it now we won't run into this issue when we write integration tests 🙂